File: q1403/q1403.java
class Solution {
    public List<Integer> minSubsequence(int[] nums) {
        Arrays.sort(nums);
        List<Integer> res = new ArrayList<>();
        int totalSum = 0;
        for (int num : nums)
            totalSum += num;

        int currSum = 0;
        for (int i = nums.length - 1; i >= 0; i--) {
            currSum += nums[i];
            res.add(nums[i]);
            if (currSum > totalSum - currSum)
                break;
        }
        return res;
    }
}

File: q1403/q1403.py
class Solution:
    def minSubsequence(self, nums: List[int]) -> List[int]:
        res = []
        nums.sort(reverse = True)
        totalSum = sum(nums)
        currSum = 0
        for num in nums:
            currSum += num
            res.append(num)
            if currSum > totalSum - currSum: 
                break
        return res